File: packages/sdk/kotlin/core/src/commonMain/kotlin/dev/edgebase/sdk/core/DbRef.kt
// EdgeBase Kotlin SDK — DbRef: namespace-scoped DB block accessor (#133 §2).
// All HTTP calls delegate to Generated Core (ApiCore.kt) via TableRef.
package dev.edgebase.sdk.core

import dev.edgebase.sdk.core.generated.GeneratedDbApi

/**
 * Reference to a DB namespace block (e.g. "shared", "workspace").
 * Use [table] to get a [TableRef] for CRUD operations.
 *
 * ```kotlin
 * val posts = client.db("shared").table("posts")
 * val rows = posts.where("status", "==", "published").get()
 * ```
 */
class DbRef(
    private val core: GeneratedDbApi,
    private val namespace: String,
    private val instanceId: String?,
    private val databaseLive: Any? // DatabaseLiveClient — Any to avoid circular dep in core
) {
    /** Get a [TableRef] for the named table. */
    @Suppress("UNCHECKED_CAST")
    fun table(name: String): TableRef {
        return TableRef(core, name, databaseLive as DatabaseLiveClient?, namespace, instanceId)
    }
}

File: packages/sdk/php/packages/core/src/DbRef.php
<?php

declare(strict_types=1);

namespace EdgeBase\Core;

use EdgeBase\Core\Generated\GeneratedDbApi;

/**
 * DbRef — reference to a DB namespace block for table access (#133 §2).
 *
 * Obtained via AdminClient::db('shared') or AdminClient::db('workspace', 'ws-456').
 *
 * Usage:
 *   $result = $admin->db('shared')->table('posts')->where('published', '==', true)->get();
 *   $result = $admin->db('workspace', 'ws-456')->table('docs')->get();
 */
class DbRef
{
    public function __construct(
        private readonly GeneratedDbApi $core,
        private readonly string $namespace,
        private readonly ?string $instanceId = null,
    ) {
    }

    /** Get a TableRef for the named table. */
    public function table(string $name): TableRef
    {
        return new TableRef($this->core, $name, $this->namespace, $this->instanceId);
    }
}
